tests/test_cadquery.py

```diff
@@ -590,6 +590,118 @@ def testSpline(self):
         self.assertAlmostEqual(path2.val().tangentAt(0).z, 0)
 
+    def testSplineWithMultipleTangents(self):
+        """
+        Tests specifying B-spline tangents, besides the start point and end
+        point tangents.
+        """
+
+        points = [(0, 0), (1, 1), (2, 0), (1, -1)]
+        tangents = [(0, 1), (1, 0), (0, -1), (-1, 0)]
+        parameters = range(len(points))
+
+        spline = (
+            Workplane("XY")
+            .spline(points, tangents=tangents, parameters=parameters)
+            .consolidateWires()
+        )
+
+        test_point = spline.edges().val().positionAt(2.5, mode="parameter")
+        expected_test_point = Vector(1.875, -0.625, 0.0)
+
+        self.assertAlmostEqual((test_point - expected_test_point).Length, 0)
+
+    def testSplineWithSpecifiedAndUnspecifiedTangents(self):
+        points = [(0, 0), (1, 1), (2, 0), (1, -1)]
+        tangents = [(0, 1), None, (0, -1), (-1, 0)]
+        parameters = range(len(points))
+
+        spline = (
+            Workplane("XY")
+            .spline(points, tangents=tangents, parameters=parameters)
+            .consolidateWires()
+        )
+
+        test_point = spline.edges().val().positionAt(1.5, mode="parameter")
+        expected_test_point = Vector(1.6875, 0.875, 0.0)
+
+        self.assertAlmostEqual((test_point - expected_test_point).Length, 0)
+
+    def testSplineSpecifyingParameters(self):
+        points = [(0, 0), (1, 1), (2, 0), (1, -1)]
+        tangents = [(0, 1), (1, 0), (0, -1), (-1, 0)]
+
+        spline1 = (
+            Workplane("XY")
+            .spline(points, tangents=tangents, parameters=[0, 1, 2, 3])
+            .consolidateWires()
+        )
+        # Multiply all parameter values by 10:
+        spline2 = (
+            Workplane("XY")
+            .spline(points, tangents=tangents, parameters=[0, 10, 20, 30])
+            .consolidateWires()
+        )
+
+        # Test point equivalence for parameter, and pamater multiplied by 10:
+        test_point1 = spline1.edges().val().positionAt(1.5, mode="parameter")
+        test_point2 = spline2.edges().val().positionAt(15, mode="parameter")
+        expected_test_point = Vector(1.625, 0.625, 0.0)
+
+        self.assertAlmostEqual((test_point1 - test_point2).Length, 0)
+        self.assertAlmostEqual((test_point1 - expected_test_point).Length, 0)
+
+    def testSplineWithScaleTrue(self):
+        points = [(0, 0), (1, 1), (2, 0), (1, -1)]
+        tangents = [(0, 1), (1, 0), (0, -1), (-1, 0)]
+        parameters = range(len(points))
+
+        spline = (
+            Workplane("XY")
+            .spline(points, tangents=tangents, parameters=parameters, scale=True)
+            .consolidateWires()
+        )
+
+        test_point = spline.edges().val().positionAt(0.5, mode="parameter")
+        expected_test_point = Vector(0.375, 0.875, 0.0)
+
+        self.assertAlmostEqual((test_point - expected_test_point).Length, 0)
```

Comment:
Hm. I think there may be some misunderstanding. The `scale` parameter doesn't have much to do with whether the tangents have unit length, or not. The length of a tangent vector has physical meaning, i.e. it changes the shape of the curve. Let me overexplain.

If your trajectory in space is a vector-valued function of a parameter time, then the tangent to your trajectory is your velocity. Compare two situations, in both of which you were at point A at some time, and at point B one second later. If your velocity at point A is in the same direction in both situations, but is 10 times larger the first time than the second time, your trajectory in the two cases is not going to be the same, even if both times your trajectory is represented by a polynomial function of the time parameter of the same degree.

Unless you have a very uniform curve (like a circle with a parameter that's the right proportion to the radius, depending on your units of length and angle), your tangent vector is not going to have a unit length.

Setting `scale = True` makes the interpolator ignore the lengths of the tangent vectors you specify, and use only their direction. The lengths themselves are interpolated.

I realized I broke up the two test cases related to the `scale` parameter. Otherwise, they use the same points, and tangent vectors. However, the test point has different coordinates at the same parameter value. That's enough to verify that `scale = True` and `scale = False` is working properly.

I'll add a commit to put the test cases together, and  add a comment.

Here's a visualization of the test cases:

___
![$Demo_spline_tangent_scaling py-spline_with_scaled_tangents](https://user-images.githubusercontent.com/1181307/108192558-6a1ae780-70e2-11eb-905d-0da03d71f5b1.png)
_Above: Spline with scaled tangents._
___
![$Demo_spline_tangent_scaling py-spline_with_unscaled_tangents](https://user-images.githubusercontent.com/1181307/108192609-7bfc8a80-70e2-11eb-9618-2bc1c4af3c6c.png)
_Above: Spline with unscaled tangents._
___
![$Demo_spline_tangent_scaling py-overlayed_splines](https://user-images.githubusercontent.com/1181307/108192660-8a4aa680-70e2-11eb-9397-42c0f8bb3a3e.png)
_Above: The two splines overlaid._
___

Reply:
> I realized I broke up the two test cases related to the `scale` parameter. Otherwise, they use the same points, and tangent vectors. However, the test point has different coordinates at the same parameter value. That's enough to verify that `scale = True` and `scale = False` is working properly.
> 
> I'll add a commit to put the test cases together, and add a comment.

Thanks, that's easier to follow now.